fineract-provider/src/main/java/org/apache/fineract/infrastructure/sms/data/SmsData.java

```diff
@@ -18,75 +18,33 @@
  */
 package org.apache.fineract.infrastructure.sms.data;
 
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import lombok.experimental.Accessors;
 import org.apache.fineract.infrastructure.core.data.EnumOptionData;
 
 /**
  * Immutable data object representing a SMS message.
  */
+
+@Data
+@NoArgsConstructor
+@Accessors(chain = true)
 public final class SmsData {
 
-    private final Long id;
-    private final Long groupId;
-    private final Long clientId;
-    private final Long staffId;
-    private final EnumOptionData status;
-    private final String mobileNo;
-    private final String message;
-    private final Long providerId;
-    private final String campaignName;
+    private Long id;
```

Comment:
Why not using `@RequiredArgsConstructor` instead ?

Reply:
@BLasan we do not want to keep `final` keyword.